### PR TITLE
Update coq

### PR DESCRIPTION
**Pull request by DanGrayson**
replaces https://github.com/UniMath/UniMath/pull/1289

### CODE REVIEW

**Comment by DanGrayson** [comment]
This sure is a slow way to learn how to run stuff inside a docker container.  I think it might be time to make a new docker image ourselves, which comes with the needed libraries already installed.  Then at least I would be able to interact with it directly and make everything run.

**Comment by benediktahrens** [comment]
```
Processing triggers for libc-bin (2.23-0ubuntu11) ...


real	0m30.871s
user	0m6.961s
sys	0m2.237s
+ rm -rf /root/.opam
+ opam --version
1.2.2
+ opam init --bare
opam: unknown option `--bare', did you mean one of `-b', `--safe', `--make'
      or `--fake' ?
Usage: opam init [OPTION]... [NAME] [ADDRESS]
Try `opam init --help' or `opam --help' for more information.
travis_time:end:004736b8:start=1570759638743270934,finish=1570759914984098180,duration=276240827246,event=script
```

**Comment by DanGrayson** [comment]
Okay, it works now, with my new docker image.  See .travis.yml for details.

The one test failure is in the sanity check for whether sub/coq was change, which it has been.

File: INSTALL_OPAM.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+`/usr/local/bin` to the `PATH` environment variable.  (Homebrew and opam both
+know how to install `ocamlc`, and we intend to use `opam` to get a version of
+`ocamlc` appropriate for compiling the version of Coq used by UniMath.)
+
+At any time, you may check that the progams installed by opam are accessible by
+you as follows.
+
+```bash
+$ type ocamlc
+ocamlc is hashed (/Users/XXXXXXXX/.opam/empty/bin/ocamlc)
```
Comment (benediktahrens) [inline INSTALL_OPAM.md]: This line seems to be specific to MacOS, despite the file being now named INSTALL_OPAM.md. Not sure it is important, though.

Comment (DanGrayson) [reply INSTALL_OPAM.md]: The idea is to document the use of opam under all systems, simultaneously.  I'll have some updates to that file anon.

**Pull request merged**